src/main/scala/io/iohk/ethereum/network/PeerEventBusActor.scala

```diff
@@ -0,0 +1,131 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.event.ActorEventBus
+import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.{MessageFromPeer, MessageToPeer, PeerDisconnected, PeerHandshakeSuccessful}
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier._
+import io.iohk.ethereum.network.handshaker.Handshaker.HandshakeResult
+import io.iohk.ethereum.network.p2p.Message
+
+object PeerEventBusActor {
+  def props: Props = Props(new PeerEventBusActor)
+
+  sealed trait PeerSelector {
+    def contains(peerId: PeerId): Boolean
+  }
+
+  object PeerSelector {
+
+    case object AllPeers extends PeerSelector {
+      override def contains(p: PeerId): Boolean = true
+    }
+
+    case class WithId(peerId: PeerId) extends PeerSelector {
+      override def contains(p: PeerId): Boolean = p == peerId
+    }
+
+  }
+
+  sealed trait SubscriptionClassifier
+
+  object SubscriptionClassifier {
+    case class MessageReceivedClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
+    case class MessageSentClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
```

Comment:
:thinking: I see your point, looking at it that way (as a `EtcPeerManager`) it would make sense for it to send the messages (as done currently with the `PeerManager.SendMessage`).
I'll take a look at it, and see if there's any problem with implementing it that way it, but it seems to make sense. I think that sending messages to the `EtcPeerManager` would also enable us to remove the use of `underlyingMsg`.